### ISSUE TITLE
Stream commands incorrect keys parsing

### ISSUE THREAD

**Issue by borsucok** [NodeRedis/redis-commands#30]
Command *xreadgroup*:

in index.js should be added:

```javascript
case 'xreadgroup':
        for (i = 0; i < args.length - 1; i++) {
          if (args[i].toUpperCase() === 'STREAMS') {
            for (var j = i + 1; j < (args.length - 1); j++) {
              keys.push(j)
            }
            break
          }
        }
      break
```

**Comment by borsucok** [comment]
See pull request https://github.com/NodeRedis/redis-commands/pull/31

**Comment by stockholmux** [comment]
Could you please describe the behaviour which is causing problems? As it's written, I'm not seeing a problem.

**Comment by borsucok** [comment]
Yes, sorry.

The problem is, that when you use ioredis, which is using this library, there is an option keyPrefix, which automaticaly adds prefix to keys in every command.

When you call xreadgroup within ioredis library, the command on redis should look like XREADGROUP GROUP groupName .... STREAMS key1

This library provides key info for first string from command, which is GROUP and when you use some prefix the command is:

`XREADGROUP prefix:GROUP groupName .... STREAMS key1`

instead of

`XREADGROUP GROUP groupName .... STREAMS prefix:key1`

**Comment by ford-prefect** [comment]
This applies to `XREAD` as well.

**Comment by voraparth91** [comment]
This patch seems to work fine. Is there any particular reason why is this not yet merged?

**Comment by borsucok** [comment]
> This patch seems to work fine. Is there any particular reason why is this not yet merged?

There were some mistakes discussed in the pr: https://github.com/NodeRedis/redis-commands/pull/31

**Comment by BridgeAR** [comment]
Fixed by 43b4efe. This is included in the latest release.
